### DISCUSSION TITLE
Unable to create User from Studio (self-hosted), "Failed to create user: invalid JWT: unable to parse or verify signature, signature is invalid" error

### DISCUSSION THREAD

**Discussion by boardmain** [supabase/supabase#19603] [Questions]
hi all i 
receive this error 
"Failed to create user: invalid JWT: unable to parse or verify signature, signature is invalid"
when i try to create a user via supabase studio ( self hosted )

where can i check to better understand to solve the error?

supabase latest version 

  container_name: supabase-studio
  image: supabase/studio:20231123-64a766a

**Comment by crackedbeefcake** [comment]
same. any progress or update? 


my gotrue service logs:
```
supabase-auth  | {"component":"api","error":"401: invalid JWT: unable to parse or verify signature, signature is invalid","level":"info","method":"POST","msg":"401: invalid JWT: unable to parse or verify signature, signature is invalid","path":"/invite","referer":"https://supa.mydomain.com","remote_addr":"172.18.0.10","time":"2024-02-19T10:02:03Z","timestamp":"2024-02-19T10:02:03Z"}
```

**Reply by lpkobamn** [reply]
any news?

**Reply by crackedbeefcake** [reply]
nope. i can not create users from supabase studio

**Reply by lpkobamn** [reply]
I reinstalled everything according to the instructions, received new tokens according to the instructions and everything worked.

**Reply by crackedbeefcake** [reply]
https://supabase.com/docs/guides/self-hosting/docker#securing-your-services
are you refering to this?
if so, then no luck for me. 

after creating a JWT SECRET and using the provided `Generate JWT` button, I still cant create users 
and when trying to use https://supa.mydomain.com/project/default/api/graphiql the playground says `{"message": "Invalid authentication credentials"}`. 

interestingly, while on the graphiql playground, kong keeps asking me over and over to log in to studio. 
also, hosting [my own graphiql](https://supabase.com/docs/guides/graphql#graphiql) works.

**Reply by Hoopcode** [reply]
Hey @LazerJesus, I think the issue is with their button. When I created the anon and service_role_key using the JWT python library, it worked fine. Below is my script

import jwt

# Payload data
payload = {
    "role": "service_role",
    "iss": "supabase",
    "iat": 1714975200,
    "exp": 1872741600
}

# Secret key (make sure to keep this secure and not hard-coded in production)
secret_key = "Your Key Here"  # Replace 'your_secret_key_here' with your actual secret key

# Generate JWT
encoded_jwt = jwt.encode(payload, secret_key, algorithm="HS256")
print("Generated JWT:", encoded_jwt)

Hope this helps!

**Reply by ghost** [reply]
@Hoopcode 
~~Can you please provide the raw code for that python script, what  I have isn't working.~~

Edit. I need the `pyjwt` library not just `jwt`

Here is the raw code.
```py
import jwt

payload_anon = {
"role": "anon",
"iss": "supabase",
"iat": 1714975200,
"exp": 1872741600
}

payload_service_role = {
"role": "service_role",
"iss": "supabase",
"iat": 1714975200,
"exp": 1872741600
}

secret = "your_jwt_secret"

encoded_anon = jwt.encode(payload_anon, secret, algorithm="HS256")
encoded_sr = jwt.encode(payload_service_role, secret, algorithm="HS256")
print("Service Key: ", encoded_sr)
print("Anon Key: ", encoded_anon)
```
